src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/AttributeDictionary.cs

```diff
@@ -0,0 +1,704 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Mvc.ViewFeatures
+{
+    /// <summary>
+    /// A dictionary for HTML attributes.
+    /// </summary>
+    public class AttributeDictionary : IDictionary<string, string>, IReadOnlyDictionary<string, string>
+    {
+        private List<KeyValuePair<string, string>> _items;
+
+        /// <inheritdoc />
+        public string this[string key]
+        {
+            get
+            {
+                if (key == null)
+                {
+                    throw new ArgumentNullException(nameof(key));
+                }
+
+                if (Count == 0)
```

Comment:
What's the current behavior when you have no entries and you attempt to look it up? I'd think it throws a `KeyNotFoundEx`